### ISSUE TITLE
[Validator] Support Global profiles from IGs

### ISSUE THREAD

**Issue by ewoutkramer** [ewoutkramer/fhir-net-api#462]
If you validate instances in the context of an IG, that IG (formally by means of ImplementationGuide.global) may specify a set of IG specific profiles that a kind of resource must conform to. I.e. although Observation.subject and many other core resources refers to Patient, in the context of a Dutch IG, all these references must be interpreted (or replaced by) references to a dutch constrained version of Patient.

Add a feature to the validator to support "profile substitution", where any reference to a core profile X, would be translated to a reference to a more specific profile from such an IG.

**Comment by wmrutten** [comment]
An IG resource can define dependencies to other IG resources:
https://www.hl7.org/fhir/implementationguide-definitions.html#ImplementationGuide.dependency

Should this logic also try to resolve the parent IG's and verify any inherited global profiles?

**Comment by ewoutkramer** [comment]
I've almost assumed these references are "informal", but I am not sure....

**Comment by wmrutten** [comment]
I think this is intended for validation purposes...

**Comment by Burnaster** [comment]
Any plans to support IGs in the near future? :)

**Comment by wmrutten** [comment]
Just thinking out loud, we could implement a new `ResourceResolver` that indexes all dependencies referenced by an IG. The resolver could also build a closure of all external dependencies to other IGs. This ResourceResolver could then be consumed by the `SnapshotGenerator` & `Validator` classes.

**Comment by ewoutkramer** [comment]
Yes, if we combine that with package support (which @mharthoorn has already built) we basically have all the tools in place we need, based on the existing framework.

**Comment by wmrutten** [comment]
Sounds like a plan! Shouldn't be too complicated.

**Comment by marcovisserFurore** [comment]
Duplicate of #1863

**Comment by ewoutkramer** [comment]
Why is this marked as a duplicate of #1863?  This issue is about the "global" feature in the IG resource, and has nothing to do with NPM packages...
